### PR TITLE
feat(secrets-nats-kv): add management CLI

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR adds a management CLI for the NATS KV secrets backend, allowing users to run the backend and separately put secrets and configure mappings.

This PR also
- Simplifies the config e2e test with new client utility functions
- Adds support to `wash-cli` for generating xkeys

## Related Issues
#2411 

A few issues should be created from this PR
- Add remove_secret utility to secrets-nats-kv
- Add get_secret utility to secrets-nats-kv
- Add release pipeline for releasing the secrets-nats-kv backend

### CODE REVIEW

File: crates/secrets-nats-kv/src/lib.rs [hunk 1]
```diff
@@ -5,4 +5,4 @@ pub mod types;
 pub use types::*;
 
 pub mod client;
-pub use client::*;
+// pub use client::*;
```
Comment (joonas) [inline crates/secrets-nats-kv/src/lib.rs:8]: Is this intentionally left commented out?

File: crates/wash-cli/src/keys.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ const NKEYS_EXTENSION: &str = ".nk";
 pub enum KeysCliCommand {
     #[clap(name = "gen", about = "Generates a keypair")]
     GenCommand {
-        /// The type of keypair to generate. May be Account, User, Module (or Component), Service (or Provider), Server (or Host), Operator, Cluster
+        /// The type of keypair to generate. May be Account, User, Module (or Component), Service (or Provider), Server (or Host), Operator, Cluster, X25519
```
Comment (joonas) [inline crates/wash-cli/src/keys.rs:18]: I believe they are called Curve in [`nk`](https://github.com/wasmCloud/nkeys/blob/main/src/bin/nk/main.rs#L26), I think using `Curve` here would be good for consistency :)

File: crates/wash-cli/src/keys.rs [hunk 3]
```diff
@@ -64,6 +64,7 @@ pub fn keytype_parser(keytype: &str) -> Result<KeyPairType> {
         "server" | "host" => Ok(KeyPairType::Server),
         "operator" => Ok(KeyPairType::Operator),
         "cluster" => Ok(KeyPairType::Cluster),
+        "x25519" => Ok(KeyPairType::Curve),
```
Comment (joonas) [inline crates/wash-cli/src/keys.rs:67]: Ditto about calling them `curve`

File: crates/secrets-nats-kv/src/lib.rs (hunk 1 above)
Comment (brooksmtownsend) [reply crates/secrets-nats-kv/src/lib.rs:8]: Nope

File: crates/wash-cli/src/keys.rs (hunk 2 above)
Comment (brooksmtownsend) [reply crates/wash-cli/src/keys.rs:18]: well look at that, looks like they both are set this way. Let me add an alias at least
```
➜ nk -gen
flag needs an argument: -gen
Usage: nk [options]
    -v                    Show version
    -gen <type>           Generate key for [user|account|server|cluster|operator|curve|x25519]
```

File: crates/host/src/wasmbus/mod.rs [hunk 4]
```diff
@@ -2998,6 +2996,8 @@ impl Host {
         provider: &Provider,
         link: &InterfaceLinkDefinition,
     ) -> anyhow::Result<()> {
+        eprintln!("resolving link {link:?}");
+        eprintln!("provider claims {:?}", provider.claims_token);
```
Comment (joonas) [inline crates/host/src/wasmbus/mod.rs:3000]: should these be `debug!(...)` instead?

File: crates/secrets-nats-kv/Dockerfile [hunk 5]
```diff
@@ -23,4 +23,4 @@ WORKDIR /app
 COPY --from=builder /app/target/release/secrets-nats-kv .
 
 # Set the entrypoint
-ENTRYPOINT ["./secrets-nats-kv"]
\ No newline at end of file
+ENTRYPOINT ["./secrets-nats-kv", "run"]
```
Comment (joonas) [inline crates/secrets-nats-kv/Dockerfile:26]:
```suggestion
ENTRYPOINT ["./secrets-nats-kv"]

CMD ["run"]
```

Comment (joonas) [reply crates/secrets-nats-kv/Dockerfile:26]: This allows folks to use the same docker image for CLI operations without having to override `entrypoint`

**Review by joonas — APPROVED** [review]
Looks good outside of the couple of nits

File: crates/host/src/wasmbus/mod.rs (hunk 4 above)
Comment (brooksmtownsend) [reply crates/host/src/wasmbus/mod.rs:3000]: oh those should be gone, thank you

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**